src/Effects.js

```diff
@@ -600,9 +600,153 @@ module.exports = class Effects {
         });
       }
     };
+
+    this.fireworks = {
+      particles:[],
+      minExplosion:20,
+      maxExplosion:50,
+      minPotential:200,
+      maxPotential:300,
+      enableTracers:true,
+      buffer:null,
+
+      makeParticle: function (type, pos, vel, color, potential) {
+        return  {
+          type:type,
+          pos:pos,
+          vel:vel,
+          gravity: p5.createVector(0.0, 0.1),
+          potential:potential,
+          acc:p5.createVector(0, 0),
+          color:color,
+          update: function () {
+            this.acc.add(this.gravity);
+            this.vel.add(this.acc);
+            this.pos.add(this.vel);
+            this.acc.mult(0, 0);
+          },
+        };
+      },
+
+      draw: function () {
+        if (this.buffer === null) {
+          this.buffer = p5;
+          // We get the tracer effect by writing frames to a
+          // offscreen buffer that has a transparent background
+          if (this.enableTracers) {
+            this.buffer = p5.createGraphics(p5.width, p5.height);
```

Comment:
@cpirich identified a memory leak in Safari when using p5.createGraphics in a background effect, documented in https://github.com/code-dot-org/dance-party/pull/26.  Can we review carefully for that?

Reply:
agreed to merge and test performance in testing / staging.